jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BaseImageParameters.java

```diff
@@ -29,13 +30,30 @@
 public class BaseImageParameters {
 
   private final AuthParameters auth;
-
-  @Nullable private String image;
   @Nullable private String credHelper;
+  @Nullable private String image;
+  private final PlatformParametersSpec platformsParametersSpec;
+  private final ListProperty<PlatformParameters> platforms;
 
   @Inject
   public BaseImageParameters(ObjectFactory objectFactory) {
     auth = objectFactory.newInstance(AuthParameters.class, "from.auth");
+    platforms = objectFactory.listProperty(PlatformParameters.class).empty();
+    platformsParametersSpec =
+        objectFactory.newInstance(PlatformParametersSpec.class, objectFactory, platforms);
+
+    PlatformParameters platform = new PlatformParameters();
+    platform.os = "linux";
+    platform.architecture = "amd64";
+    platforms.add(platform);
+  }
+
```

Comment:
You need to add some annotations like [this](https://github.com/GoogleContainerTools/jib/blob/00dbfee080e66c238bd1f1168205cab9e96e11b7/jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/JibExtension.java#L218-L219). For example, `@Optional` tells the Gradle system that `from.platforms` is an optional config that the user doesn't have to specify.